Remove iOS 13 Widgets

mokagio opened the pull request:
Removes the iOS 13 widgets, WordPressTodayWidget, WordPressThisWeekWidget, and WordPressAllTimeWidget, plush does as little work as possible to get the app to build with iOS 14 as the minimum deployment target (#19496).

See also commit messages for additional details on the rationale behind some of the changes.

While working on this, I addressed a few more deprecations, see #19512, but I decided to keep them separated to focus this PR on widgets.

## Regression Notes

1. Potential unintended areas of impact – N.A.
2. What I did to test those areas of impact (or what existing automated tests I relied on) – N.A.
3. What automated tests I added (or what prevented me from doing so) – N.A.

---

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes. **N.A.**
- [x] I have considered adding accessibility improvements for my changes. **N.A.**
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary. **N.A.**

mokagio committed:
Remove target-level `IPHONEOS_DEPLOYMENT_TARGET` setting

This way, all the targets will use the same iOS deployment, all coming
from the project-level setting.

mokagio committed:
Extract `IPHONEOS_DEPLOYMENT_TARGET` to `SDKVersions.xcconfig`

mokagio committed:
Bump `IPHONEOS_DEPLOYMENT_TARGET` from 13.0 to 14.0

mokagio committed:
Add `RELEASE-NOTES.txt` entry for dropping iOS 13

mokagio committed:
Remove `WordPressTodayWidget` in preparation to drop iOS 13

This legacy widget extension has been superseded by
`WordPressStatsWidgets`.

mokagio committed:
Remove bundle identifiers for iOS 13 widgets in Jetpack from `Fastfile`

These were there to be ready to use once the team was ready to add the
widgets. In the meantime, we decided to drop iOS 13, which removes the
need for them.

mokagio committed:
Remove leftover references to old `WordPressHomeWidgetToday`

This would have been the name of the very first iOS 10 home widget we
added. Since then, it was renamed but we missed the dead references.

mokagio commented on WordPress/Info.plist:
```diff
@@ -550,8 +550,6 @@
 		<string>Nunito-Bold.ttf</string>
 		<string>EBGaramond-Regular.ttf</string>
 	</array>
-	<key>UIApplicationShortcutWidget</key>
-	<string>org.wordpress.WordPressTodayWidget</string>
```
As far as I could gather from [the docs](https://developer.apple.com/documentation/bundleresources/information_property_list/uiapplicationshortcutwidget) and the latest [support page](https://support.apple.com/en-us/HT207122), this key was used at the time of the iOS 10 to 13 widgets but is no longer needed.

mokagio commented on WordPress/WordPress.xcodeproj/project.pbxproj:
```diff
@@ -5593,13 +5461,10 @@
 		09C8BB8327DFF9CB00974175 /* ro */ = {isa = PBXFileReference; lastKnownFileType = text.plist.strings; name = ro; path = ro.lproj/InfoPlist.strings; sourceTree = "<group>"; };
 		09C8BB8427DFF9CC00974175 /* sk */ = {isa = PBXFileReference; lastKnownFileType = text.plist.strings; name = sk; path = sk.lproj/InfoPlist.strings; sourceTree = "<group>"; };
 		09F367D2BE684EDE2E4A40E3 /* Pods-WordPressDraftActionExtension.debug.xcconfig */ = {isa = PBXFileReference; includeInIndex = 1; lastKnownFileType = text.xcconfig; name = "Pods-WordPressDraftActionExtension.debug.xcconfig"; path = "../Pods/Target Support Files/Pods-WordPressDraftActionExtension/Pods-WordPressDraftActionExtension.debug.xcconfig"; sourceTree = "<group>"; };
-		0A27D496128FBCDFA3658994 /* Pods-WordPressHomeWidgetToday.release-internal.xcconfig */ = {isa = PBXFileReference; includeInIndex = 1; lastKnownFileType = text.xcconfig; name = "Pods-WordPressHomeWidgetToday.release-internal.xcconfig"; path = "../Pods/Target Support Files/Pods-WordPressHomeWidgetToday/Pods-WordPressHomeWidgetToday.release-internal.xcconfig"; sourceTree = "<group>"; };
```
Fun fact. We've been carrying around these `WordPressHomeWidgetToday` dead references ever since renaming that target to `WordPressTodayWidget` 😳

staskus replied:
Thanks for noticing. The project will get a well-deserved clean-up then  👀

mokagio committed:
Use `CLLocationManager` instance property for `authorizationStatus`

The class property was deprecated in iOS 14. Xcode gave the following
warning (which we treat as an error):

>  Deprecated. Use the instance property `authorizationStatus` instead.

mokagio committed:
Replace deprecated `CLLocationManagerDelegate` method

The deprecation warning in Xcode wasn't informative, but the deprecation
annotation suggested what to do:

```
/*
 *  locationManager:didChangeAuthorizationStatus:
 *
 *  Discussion:
 *    Invoked when the authorization status changes for this application.
 */
- (void)locationManager:(CLLocationManager *)manager didChangeAuthorizationStatus:(CLAuthorizationStatus)status API_DEPRECATED_WITH_REPLACEMENT("-locationManagerDidChangeAuthorization:", ios(4.2, 14.0), macos(10.7, 11.0), watchos(1.0, 7.0), tvos(9.0, 14.0));
```

mokagio committed:
Remove iOS 13 only `SchedulingCalendarViewController`

mokagio commented:
cc @crazytonyli, if you have the time to take a look.

staskus reviewed (commented):
Thank you for the changes, @mokagio! I reviewed commit by commit. Additional changes unrelated to Widgets make sense as well. 👍
 
⚠️ Additional files that can be removed (In Today Widgets / Shared Views):
- `WidgetDiffferenceCell`
- `WidgetTwoCollumnCell`
- `WidgetUnconfiguredCell`
- `WidgetUrlCell`
- `WidgetNoConnectionCell`

Potentially, there could be minor details related to iOS13 Widgets that could be removed as well, for example, some Constants and details saved to UserDefaults. However, iOS13 Widget data-loading code is reused by iOS14+ Widgets, therefore, it's hard to tell if any small details could be removed without negatively affecting current widgets. The majority of the code is removed in this PR. 

⚠️ I see there're multiple places that still have `#available(iOS 14.0, *)`, including Stats-related `StatsWidgetStore`. However, they could be removed in the parent PR or later. 

I tested the app:
- iOS 14 widgets continue working
- Existing shortcuts continue working

mokagio committed:
Remove `WordPressAllTimeWidget` in preparation to drop iOS 13

This legacy widget extension has been superseded by
`WordPressStatsWidgets`.

mokagio committed:
Remove `WordPressThisWeekWidget` in preparation to drop iOS 13

This legacy widget extension has been superseded by
`WordPressStatsWidgets`.

mokagio committed:
Remove `WidgetDifferenceCell` — It was used only in the iOS 13 widgets

mokagio committed:
Remove `WidgetTwoColumnCell` — It was used only in the iOS 13 widgets

mokagio committed:
Remove `WidgetUnconfiguredCell` — It was used only in the iOS 13 widgets

mokagio committed:
Remove `WidgetUrlCell` — It was used only in the iOS 13 widgets

mokagio committed:
Remove `WidgetNoConnectionCell` — It was used only in the iOS 13 widgets

mokagio committed:
Remove empty `Shared View` group for widgets

mokagio commented:
Thank you for the review @staskus 🙌 

I removed the Shared Views that this work made unused as you suggested.

> I see there're multiple places that still have `#available(iOS 14.0, *)`, including Stats-related `StatsWidgetStore`. However, they could be removed in the parent PR or later.

Agreed. I might leave them for a PR later on. As a matter of fact, I already removed them [here](https://github.com/wordpress-mobile/WordPress-iOS/pull/19512/commits/3069cb7fd81be9d130ddf444c58208e15c3b50d1) but I think I'll submit them in isolation after #19496 lands.

staskus reviewed (commented):
@mokagio, [I don't see SharedViews being removed](https://github.com/wordpress-mobile/WordPress-iOS/tree/mokagio/remove-ios-13-widgets/WordPress/Classes/ViewRelated/Stats/Today%20Widgets/Shared%20Views).

staskus reviewed (approved):
I don't want to block this PR, so after shared views are removed, we can move on. 👍

mokagio commented:
> @mokagio, [I don't see SharedViews being removed](https://github.com/wordpress-mobile/WordPress-iOS/tree/mokagio/remove-ios-13-widgets/WordPress/Classes/ViewRelated/Stats/Today%20Widgets/Shared%20Views?rgh-link-date=2022-10-27T06%3A41%3A31Z).

I didn't push them... 😓 But you can see them in the timeline now

<img width="1044" alt="image" src="https://user-images.githubusercontent.com/1218433/198416805-f54ba188-b6df-4977-9fc5-9bc4662daa24.png">

Thank you for the approval to unblock. I'll merge as soon as CI is green.

staskus commented:
I see the changes now, all good 🚀

guarani reviewed (approved):
Apologies for the late review, @mokagio. The code changes look right and I gave the Jetpack app and the WP app both a quick smoke test (adding Home Screen widgets, scheduling a post, etc.) – all good.

I love how this reduced the number of targets in the project ❤️  Thanks for wrangling this!

Are you planning to merge this in 21.1 or bump to the next version?

mokagio committed:
Update `RELEASE-NOTES.txt` to mention new PR for dropping iOS 13
